In [1]:
import pandas as pd
import numpy as np

In [2]:
from pymongo import MongoClient

In [3]:
client= MongoClient('mongodb://localhost:27017/')

In [4]:
mydatabase= client['intern']

In [5]:
collection= mydatabase['sensor_data']

In [6]:
agg_result= collection.aggregate(
[{
    "$group":
    {
        "_id":"$sensor_name",
        "Total Records":{"$sum":1}
    }
}]
)

In [7]:
for i in agg_result:
    print(i)

{'_id': 'Pump Shaft Speed', 'Total Records': 730}
{'_id': 'Pump Suction Temperature', 'Total Records': 730}
{'_id': 'Pump Thrust Bearing Temperature 1', 'Total Records': 730}
{'_id': 'Pump Thrust Bearing Temperature 2', 'Total Records': 730}
{'_id': 'Heat Recovery System Header Mass Flow', 'Total Records': 730}
{'_id': 'Auxiliary Boilers Feed Water Header Pressure', 'Total Records': 730}
{'_id': 'Motor Current Phase C', 'Total Records': 730}
{'_id': 'Motor Input Power', 'Total Records': 730}
{'_id': 'Motor Voltage', 'Total Records': 730}
{'_id': 'Lube Oil Cooler Outlet Temperature', 'Total Records': 730}
{'_id': 'Pump Journal 1 Bearing Temperature', 'Total Records': 730}
{'_id': 'Pump Suction Pressure 2', 'Total Records': 730}
{'_id': 'Pump Suction Strainer Differential Pressure', 'Total Records': 730}
{'_id': 'Heat Recovery System Header Mass Flow.1', 'Total Records': 730}
{'_id': 'Heat Recovery System Header Pressure.1', 'Total Records': 730}
{'_id': 'Pump Discharge Volumetric Flow',

In [8]:
collection.find_one()

{'_id': ObjectId('62cd515114ccc4c83c3f5f5e'),
 'start_epoch': 1514764815.0,
 'end_epoch': 1514851200.0,
 'sensor_name': 'Auxiliary Boilers A/B Feed Water Header Pressure 2',
 'data': [{'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 15),
   'value': 1305.491089},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 30), 'value': 1301.227783},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 45), 'value': 1296.178589},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1), 'value': 1297.628296},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 15), 'value': 1299.078003},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 30), 'value': 1300.527832},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 45), 'value': 1299.512085},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2), 'value': 1297.239136},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 15), 'value': 1294.966064},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 30), 'value': 1286.775269},
  {'timestamp': datetim

In [9]:
agg_result= collection.aggregate(
[{
    "$group":
    {
        "_id":"$value.timestamp",
        "Total Records":{"$sum":1}
    }
}]
)

In [10]:
for i in agg_result:
    print(i)

{'_id': None, 'Total Records': 18980}


In [11]:
records= collection.find()

In [12]:
records[0]

{'_id': ObjectId('62cd515114ccc4c83c3f5f5e'),
 'start_epoch': 1514764815.0,
 'end_epoch': 1514851200.0,
 'sensor_name': 'Auxiliary Boilers A/B Feed Water Header Pressure 2',
 'data': [{'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 15),
   'value': 1305.491089},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 30), 'value': 1301.227783},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 45), 'value': 1296.178589},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1), 'value': 1297.628296},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 15), 'value': 1299.078003},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 30), 'value': 1300.527832},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 45), 'value': 1299.512085},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2), 'value': 1297.239136},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 15), 'value': 1294.966064},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 30), 'value': 1286.775269},
  {'timestamp': datetim

In [13]:
records[1]

{'_id': ObjectId('62cd515114ccc4c83c3f5f5f'),
 'start_epoch': 1514764815.0,
 'end_epoch': 1514851200.0,
 'sensor_name': 'Auxiliary Boilers Feed Water Header Pressure',
 'data': [{'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 15),
   'value': 1305.893799},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 30), 'value': 1299.925171},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 45), 'value': 1294.646606},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1), 'value': 1299.98877},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 15), 'value': 1300.749023},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 30), 'value': 1299.165527},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 45), 'value': 1297.582031},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2), 'value': 1295.998657},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 15), 'value': 1294.415161},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 30), 'value': 1292.358276},
  {'timestamp': datetime.datet

In [14]:
records[25]

{'_id': ObjectId('62cd515114ccc4c83c3f5f77'),
 'start_epoch': 1514764815.0,
 'end_epoch': 1514851200.0,
 'sensor_name': 'Pump Thrust Bearing Temperature 2',
 'data': [{'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 15),
   'value': 86.57155609},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 30), 'value': 86.5650177},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 45), 'value': 86.55848694},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1), 'value': 86.55195618},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 15), 'value': 86.54542542},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 30), 'value': 86.53889465},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 1, 45), 'value': 86.53236389},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2), 'value': 86.5258255},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 15), 'value': 86.51929474},
  {'timestamp': datetime.datetime(2018, 1, 1, 0, 2, 30), 'value': 86.51276398},
  {'timestamp': datetime.datetime(2018, 1,

In [15]:
records[10000]

{'_id': ObjectId('62cd5a8b14ccc4c83c3f866e'),
 'start_epoch': 1547942415.0,
 'end_epoch': 1548028800.0,
 'sensor_name': 'Pump Journal 1 Bearing Temperature',
 'data': [{'timestamp': datetime.datetime(2019, 1, 20, 0, 0, 15),
   'value': 130.0657654},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 0, 30),
   'value': 130.0632935},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 0, 45),
   'value': 130.0608063},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 1), 'value': 130.0583344},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 1, 15),
   'value': 130.0558624},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 1, 30),
   'value': 130.0533905},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 1, 45),
   'value': 130.0509186},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 2), 'value': 130.0484467},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 2, 15),
   'value': 130.0459747},
  {'timestamp': datetime.datetime(2019, 1, 20, 0, 2, 30),
   'value': 130.0434875},
  {'time

In [16]:
records[18000]

{'_id': ObjectId('62cd5e4c14ccc4c83c3fa5ae'),
 'start_epoch': 1574553615.0,
 'end_epoch': 1574640000.0,
 'sensor_name': 'Motor Current Phase A',
 'data': [{'timestamp': datetime.datetime(2019, 11, 24, 0, 0, 15),
   'value': 152.63986210000004},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 0, 30), 'value': 153.0},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 0, 45),
   'value': 152.6385193},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 1), 'value': 152.0},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 1, 15),
   'value': 152.7263031},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 1, 30),
   'value': 152.73011780000004},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 1, 45),
   'value': 152.35913090000005},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 2), 'value': 153.0},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 2, 15),
   'value': 152.3605957},
  {'timestamp': datetime.datetime(2019, 11, 24, 0, 2, 30),
   'value': 152.6372681},
  {'timesta

In [17]:
records[18979]

{'_id': ObjectId('62cd5eab14ccc4c83c3fa981'),
 'start_epoch': 1577750415.0,
 'end_epoch': 1577836785.0,
 'sensor_name': 'Pump Thrust Bearing Temperature 2',
 'data': [{'timestamp': datetime.datetime(2019, 12, 31, 0, 0, 15),
   'value': 90.1043396},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 0, 30),
   'value': 90.11571503},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 0, 45),
   'value': 90.12708282},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 1), 'value': 90.13845825},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 1, 15),
   'value': 90.14983368},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 1, 30),
   'value': 90.16120911},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 1, 45),
   'value': 90.1725769},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 2), 'value': 90.18395233},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 2, 15),
   'value': 90.19532776},
  {'timestamp': datetime.datetime(2019, 12, 31, 0, 2, 30),
   'value': 90.20670319},
 

In [18]:
df=pd.DataFrame()

In [19]:
agg_result= collection.aggregate(
[{
    "$group":
    {
        "_id":"$sensor_name",
        
    }
}]
)
columns=[]
for i in agg_result:
    print(i)
    columns.append(i['_id'])

{'_id': 'Motor Current Phase C'}
{'_id': 'Motor Input Power'}
{'_id': 'Auxiliary Boilers Feed Water Header Pressure'}
{'_id': 'Heat Recovery System Header Mass Flow'}
{'_id': 'Pump Suction Pressure 1'}
{'_id': 'Pump Thrust Bearing Temperature 2'}
{'_id': 'Pump Thrust Bearing Temperature 1'}
{'_id': 'Pump Suction Temperature'}
{'_id': 'Pump Journal 2 Bearing Temperature'}
{'_id': 'Heat Recovery System Header Mass Flow.1'}
{'_id': 'Heat Recovery System Header Pressure.1'}
{'_id': 'Pump Discharge Volumetric Flow'}
{'_id': 'Pump Radial Bearing Vibration'}
{'_id': 'Lube Oil Cooler Outlet Temperature'}
{'_id': 'Pump Suction Pressure 2'}
{'_id': 'Motor Voltage'}
{'_id': 'Pump Journal 1 Bearing Temperature'}
{'_id': 'Pump Suction Strainer Differential Pressure'}
{'_id': 'Pump Discharge Pressure'}
{'_id': 'Motor Power Factor'}
{'_id': 'Auxiliary Boilers A/B Feed Water Header Pressure 2'}
{'_id': 'Pump Shaft Speed'}
{'_id': 'Lube Oil Tank Temperature'}
{'_id': 'Motor Current Phase B'}
{'_id': 'H

In [20]:
columns

['Motor Current Phase C',
 'Motor Input Power',
 'Auxiliary Boilers Feed Water Header Pressure',
 'Heat Recovery System Header Mass Flow',
 'Pump Suction Pressure 1',
 'Pump Thrust Bearing Temperature 2',
 'Pump Thrust Bearing Temperature 1',
 'Pump Suction Temperature',
 'Pump Journal 2 Bearing Temperature',
 'Heat Recovery System Header Mass Flow.1',
 'Heat Recovery System Header Pressure.1',
 'Pump Discharge Volumetric Flow',
 'Pump Radial Bearing Vibration',
 'Lube Oil Cooler Outlet Temperature',
 'Pump Suction Pressure 2',
 'Motor Voltage',
 'Pump Journal 1 Bearing Temperature',
 'Pump Suction Strainer Differential Pressure',
 'Pump Discharge Pressure',
 'Motor Power Factor',
 'Auxiliary Boilers A/B Feed Water Header Pressure 2',
 'Pump Shaft Speed',
 'Lube Oil Tank Temperature',
 'Motor Current Phase B',
 'Heat Recovery System Header Pressure',
 'Motor Current Phase A']

In [21]:
records[0]['data'][0]

{'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 15), 'value': 1305.491089}

In [22]:
records[0]['data'][1]

{'timestamp': datetime.datetime(2018, 1, 1, 0, 0, 30), 'value': 1301.227783}

In [23]:
type(records[0]['data'][1]['timestamp'])

datetime.datetime

In [24]:
records[0]['data'][1]['timestamp']

datetime.datetime(2018, 1, 1, 0, 0, 30)

In [25]:
index_val=[]

In [26]:
type(index_val)

list

In [27]:
for i in range(5760):
    index_val.append(records[0]['data'][i]['timestamp'])

In [28]:
index_val

[datetime.datetime(2018, 1, 1, 0, 0, 15),
 datetime.datetime(2018, 1, 1, 0, 0, 30),
 datetime.datetime(2018, 1, 1, 0, 0, 45),
 datetime.datetime(2018, 1, 1, 0, 1),
 datetime.datetime(2018, 1, 1, 0, 1, 15),
 datetime.datetime(2018, 1, 1, 0, 1, 30),
 datetime.datetime(2018, 1, 1, 0, 1, 45),
 datetime.datetime(2018, 1, 1, 0, 2),
 datetime.datetime(2018, 1, 1, 0, 2, 15),
 datetime.datetime(2018, 1, 1, 0, 2, 30),
 datetime.datetime(2018, 1, 1, 0, 2, 45),
 datetime.datetime(2018, 1, 1, 0, 3),
 datetime.datetime(2018, 1, 1, 0, 3, 15),
 datetime.datetime(2018, 1, 1, 0, 3, 30),
 datetime.datetime(2018, 1, 1, 0, 3, 45),
 datetime.datetime(2018, 1, 1, 0, 4),
 datetime.datetime(2018, 1, 1, 0, 4, 15),
 datetime.datetime(2018, 1, 1, 0, 4, 30),
 datetime.datetime(2018, 1, 1, 0, 4, 45),
 datetime.datetime(2018, 1, 1, 0, 5),
 datetime.datetime(2018, 1, 1, 0, 5, 15),
 datetime.datetime(2018, 1, 1, 0, 5, 30),
 datetime.datetime(2018, 1, 1, 0, 5, 45),
 datetime.datetime(2018, 1, 1, 0, 6),
 datetime.dateti

#we have a collection of 18980 documents which has values of 26 sensors (730 epochs)
#each document has another set of 5760 array documents consisting of timestamp and value of the respective sensor at that moment
#we have data from date 1:1:2018 - 31:12:2019
#Our nest steps will be to extract all the data in the database and examine them.


In [29]:
sensor= records[0]['sensor_name']
sensor

'Auxiliary Boilers A/B Feed Water Header Pressure 2'

In [30]:
#Accessing data value along with timestamp
for i in range(10):
    print(records[0]['data'][i]['timestamp'],"value:",records[0]['data'][i]['value'])

2018-01-01 00:00:15 value: 1305.491089
2018-01-01 00:00:30 value: 1301.227783
2018-01-01 00:00:45 value: 1296.178589
2018-01-01 00:01:00 value: 1297.628296
2018-01-01 00:01:15 value: 1299.078003
2018-01-01 00:01:30 value: 1300.527832
2018-01-01 00:01:45 value: 1299.512085
2018-01-01 00:02:00 value: 1297.239136
2018-01-01 00:02:15 value: 1294.966064
2018-01-01 00:02:30 value: 1286.775269


In [31]:
j=1
for i in range(27):
    
    print(records[i]['sensor_name'], "position:",j)
    j=j+1
    

Auxiliary Boilers A/B Feed Water Header Pressure 2 position: 1
Auxiliary Boilers Feed Water Header Pressure position: 2
Heat Recovery System Header Mass Flow position: 3
Heat Recovery System Header Mass Flow.1 position: 4
Heat Recovery System Header Pressure position: 5
Heat Recovery System Header Pressure.1 position: 6
Lube Oil Cooler Outlet Temperature position: 7
Lube Oil Tank Temperature position: 8
Motor Current Phase A position: 9
Motor Current Phase B position: 10
Motor Current Phase C position: 11
Motor Input Power position: 12
Motor Power Factor position: 13
Motor Voltage position: 14
Pump Discharge Pressure position: 15
Pump Discharge Volumetric Flow position: 16
Pump Journal 1 Bearing Temperature position: 17
Pump Journal 2 Bearing Temperature position: 18
Pump Radial Bearing Vibration position: 19
Pump Shaft Speed position: 20
Pump Suction Pressure 1 position: 21
Pump Suction Pressure 2 position: 22
Pump Suction Strainer Differential Pressure position: 23
Pump Suction Tempe

In [32]:
j=1
n=0
for i in range(1000):
    if records[i]['sensor_name']== columns[0]:
        
        print(j,"distance from last occurance: ",j-n)
        n=j
    j=j+1

11 distance from last occurance:  11
37 distance from last occurance:  26
63 distance from last occurance:  26
89 distance from last occurance:  26
115 distance from last occurance:  26
141 distance from last occurance:  26
167 distance from last occurance:  26
193 distance from last occurance:  26
219 distance from last occurance:  26
245 distance from last occurance:  26
271 distance from last occurance:  26
297 distance from last occurance:  26
323 distance from last occurance:  26
349 distance from last occurance:  26
375 distance from last occurance:  26
401 distance from last occurance:  26
427 distance from last occurance:  26
453 distance from last occurance:  26
479 distance from last occurance:  26
505 distance from last occurance:  26
531 distance from last occurance:  26
557 distance from last occurance:  26
583 distance from last occurance:  26
609 distance from last occurance:  26
635 distance from last occurance:  26
661 distance from last occurance:  26
687 distance fro

#checked and found the sensors are placed in an exact order throughout the collection. we can change the value of columns[i=(0,25) and check it]

will use pandas series append to make panda dateframe then use a csv file to store them


In [33]:
print(records[0]['sensor_name'])
for i in range(5760):
    s=(records[0]['data'][i]['value'])


Auxiliary Boilers A/B Feed Water Header Pressure 2


#Next we will insert value for every sensor as a series into our dataframe

In [34]:
df=pd.DataFrame()
index_val=[]
for i in range(5760):  #no of entries in data array to extract timestamps
    index_val.append(records[0]['data'][i]['timestamp'])
df.index=index_val
for variables in range(26): # extracting value for each of 26 variables
    for timestamp in range(5760): # extracting value of each variable for each timestamp
        s=(records[variables]['data'][timestamp]['value'])
    df.insert(variables,column=records[variables]['sensor_name'],value=s)

In [35]:
df

Auxiliary Boilers A/B Feed Water Header Pressure 2  \
2018-01-01 00:00:15                                        1292.691772    
2018-01-01 00:00:30                                        1292.691772    
2018-01-01 00:00:45                                        1292.691772    
2018-01-01 00:01:00                                        1292.691772    
2018-01-01 00:01:15                                        1292.691772    
...                                                                ...    
2018-01-01 23:59:00                                        1292.691772    
2018-01-01 23:59:15                                        1292.691772    
2018-01-01 23:59:30                                        1292.691772    
2018-01-01 23:59:45                                        1292.691772    
2018-01-02 00:00:00                                        1292.691772    

                     Auxiliary Boilers Feed Water Header Pressure  \
2018-01-01 00:00:15                                   1288.269897   
2018-01-01 00:00:30                                   1288.269897   
2018-01-01 00:00:45                                   1288.269897   
2018-01-01 00:01:00                                   1288.269897   
2018-01-01 00:01:15                                   1288.269897   
...                                                           ...   
2018-01-01 23:59:00                                   1288.269897   
2018-01-01 23:59:15                                   1288.269897   
2018-01-01 23:59:30                                   1288.269897   
2018-01-01 23:59:45                                   1288.269897   
2018-01-02 00:00:00                                   1288.269897   

                     Heat Recovery System Header Mass Flow  \
2018-01-01 00:00:15                            304351.9063   
2018-01-01 00:00:30                            304351.9063   
2018-01-01 00:00:45                            304351.9063   
2018-01-01 00:01:00                            304351.9063   
2018-01-01 00:01:15                            304351.9063   
...                                                    ...   
2018-01-01 23:59:00                            304351.9063   
2018-01-01 23:59:15                            304351.9063   
2018-01-01 23:59:30                            304351.9063   
2018-01-01 23:59:45                            304351.9063   
2018-01-02 00:00:00                            304351.9063   

                     Heat Recovery System Header Mass Flow.1  \
2018-01-01 00:00:15                              307503.0313   
2018-01-01 00:00:30                              307503.0313   
2018-01-01 00:00:45                              307503.0313   
2018-01-01 00:01:00                              307503.0313   
2018-01-01 00:01:15                              307503.0313   
...                                                      ...   
2018-01-01 23:59:00                              307503.0313   
2018-01-01 23:59:15                              307503.0313   
2018-01-01 23:59:30                              307503.0313   
2018-01-01 23:59:45                              307503.0313   
2018-01-02 00:00:00                              307503.0313   

                     Heat Recovery System Header Pressure  \
2018-01-01 00:00:15                           1755.683594   
2018-01-01 00:00:30                           1755.683594   
2018-01-01 00:00:45                           1755.683594   
2018-01-01 00:01:00                           1755.683594   
2018-01-01 00:01:15                           1755.683594   
...                                                   ...   
2018-01-01 23:59:00                           1755.683594   
2018-01-01 23:59:15                           1755.683594   
2018-01-01 23:59:30                           1755.683594   
2018-01-01 23:59:45                           1755.683594   
2018-01-02 00:00:00                           1755.683594   

                     Heat Recovery System He

In [36]:
df.describe()

Auxiliary Boilers A/B Feed Water Header Pressure 2  \
count                                       5.760000e+03    
mean                                        1.292692e+03    
std                                         5.457442e-11    
min                                         1.292692e+03    
25%                                         1.292692e+03    
50%                                         1.292692e+03    
75%                                         1.292692e+03    
max                                         1.292692e+03    

       Auxiliary Boilers Feed Water Header Pressure  \
count                                  5.760000e+03   
mean                                   1.288270e+03   
std                                    5.457442e-11   
min                                    1.288270e+03   
25%                                    1.288270e+03   
50%                                    1.288270e+03   
75%                                    1.288270e+03   
max                                    1.288270e+03   

       Heat Recovery System Header Mass Flow  \
count                           5.760000e+03   
mean                            3.043519e+05   
std                             1.228288e-08   
min                             3.043519e+05   
25%                             3.043519e+05   
50%                             3.043519e+05   
75%                             3.043519e+05   
max                             3.043519e+05   

       Heat Recovery System Header Mass Flow.1  \
count                             5.760000e+03   
mean                              3.075030e+05   
std                               1.286501e-08   
min                               3.075030e+05   
25%                               3.075030e+05   
50%                               3.075030e+05   
75%                               3.075030e+05   
max                               3.075030e+05   

       Heat Recovery System Header Pressure  \
count                          5.760000e+03   
mean                           1.755684e+03   
std                            2.217086e-10   
min                            1.755684e+03   
25%                            1.755684e+03   
50%                            1.755684e+03   
75%                            1.755684e+03   
max                            1.755684e+03   

       Heat Recovery System Header Pressure.1  \
count                            5.760000e+03   
mean                             1.759273e+03   
std                              9.414087e-11   
min                              1.759273e+03   
25%                              1.759273e+03   
50%                              1.759273e+03   
75%                              1.759273e+03   
max                              1.759273e+03   

       Lube Oil Cooler Outlet Temperature  Lube Oil Tank Temperature  \
count                        5.760000e+03               5.760000e+03   
mean                         6.679857e+01               7.272994e+01   
std                          5.755896e-12               9.109949e-12   
min                          6.679857e+01               7.272994e+01   
25%                          6.679857e+01               7.272994e+01   
50%                          6.679857e+01               7.272994e+01   
75%                          6.679857e+01               7.272994e+01   
max                          6.679857e+01               7.272994e+01   

       Motor Current Phase A  Motor Current Phase B  ...  \
count           5.760000e+03           5.760000e+03  ...   
mean            1.503091e+02           1.503099e+02  ...   
std             1.665657e-11           1.037482e-11  ...   
min             1.503091e+02           1.503099e+02  ...   
25%             1.503091e+02           1.503099e+02  ...   
50%             1.503091e+02           1.503099e+02  ...   
75%             1.503091e+02           1.503099e+02  ...   
max             1.503091e+02           1.503099e+02  ...   

       Pump Journal 1

In [37]:
df.isna().sum()

Auxiliary Boilers A/B Feed Water Header Pressure 2       0
Auxiliary Boilers Feed Water Header Pressure             0
Heat Recovery System Header Mass Flow                    0
Heat Recovery System Header Mass Flow.1                  0
Heat Recovery System Header Pressure                     0
Heat Recovery System Header Pressure.1                   0
Lube Oil Cooler Outlet Temperature                       0
Lube Oil Tank Temperature                                0
Motor Current Phase A                                    0
Motor Current Phase B                                    0
Motor Current Phase C                                    0
Motor Input Power                                     5760
Motor Power Factor                                    5760
Motor Voltage                                            0
Pump Discharge Pressure                                  0
Pump Discharge Volumetric Flow                           0
Pump Journal 1 Bearing Temperature                      

In [38]:
df.to_csv("epoch1.csv")

#as the standard deviation of each fielsd is negligible we are going to take the mean from ech poch set to make the trainng data manageable

In [39]:
agg_result= collection.aggregate(
[{ 
    "$project":
    {
        "_id":"$sensor_name",
        "value":{"$avg":"$data.value"}
    }
}]
)

In [40]:
count=0
for i in agg_result:
    print(i)
    count=count+1
print("number of records to be considered:", count)

{'_id': 'Auxiliary Boilers A/B Feed Water Header Pressure 2', 'value': 1299.9998471440972}
{'_id': 'Auxiliary Boilers Feed Water Header Pressure', 'value': 1299.9323672413193}
{'_id': 'Heat Recovery System Header Mass Flow', 'value': 302136.6891086111}
{'_id': 'Heat Recovery System Header Mass Flow.1', 'value': 305769.2011641493}
{'_id': 'Heat Recovery System Header Pressure', 'value': 1772.3438652251737}
{'_id': 'Heat Recovery System Header Pressure.1', 'value': 1774.8517823329862}
{'_id': 'Lube Oil Cooler Outlet Temperature', 'value': 68.59462388620312}
{'_id': 'Lube Oil Tank Temperature', 'value': 74.3533583256632}
{'_id': 'Motor Current Phase A', 'value': 147.28868712335068}
{'_id': 'Motor Current Phase B', 'value': 147.58546372569447}
{'_id': 'Motor Current Phase C', 'value': 146.02334957430554}
{'_id': 'Motor Input Power', 'value': nan}
{'_id': 'Motor Power Factor', 'value': nan}
{'_id': 'Motor Voltage', 'value': 4236.037884018403}
{'_id': 'Pump Discharge Pressure', 'value': 1783

{'_id': 'Heat Recovery System Header Pressure.1', 'value': 1731.2823530864584}
{'_id': 'Lube Oil Cooler Outlet Temperature', 'value': 77.9128259645}
{'_id': 'Lube Oil Tank Temperature', 'value': 79.36406300729166}
{'_id': 'Motor Current Phase A', 'value': 153.34447801144097}
{'_id': 'Motor Current Phase B', 'value': 153.58665299461805}
{'_id': 'Motor Current Phase C', 'value': 152.38170444717014}
{'_id': 'Motor Input Power', 'value': nan}
{'_id': 'Motor Power Factor', 'value': nan}
{'_id': 'Motor Voltage', 'value': 4231.079357075347}
{'_id': 'Pump Discharge Pressure', 'value': 1742.2988274699653}
{'_id': 'Pump Discharge Volumetric Flow', 'value': 962.5693412248784}
{'_id': 'Pump Journal 1 Bearing Temperature', 'value': 140.58171824020832}
{'_id': 'Pump Journal 2 Bearing Temperature', 'value': 145.99100981032987}
{'_id': 'Pump Radial Bearing Vibration', 'value': 8.57394630280573}
{'_id': 'Pump Shaft Speed', 'value': 3588.0790261085067}
{'_id': 'Pump Suction Pressure 1', 'value': 24.4722

{'_id': 'Pump Suction Strainer Differential Pressure', 'value': 0.32465731539027776}
{'_id': 'Pump Suction Temperature', 'value': 226.67831523899306}
{'_id': 'Pump Thrust Bearing Temperature 1', 'value': 108.84143222527779}
{'_id': 'Pump Thrust Bearing Temperature 2', 'value': 99.8472528453073}
{'_id': 'Auxiliary Boilers A/B Feed Water Header Pressure 2', 'value': 1300.276848503993}
{'_id': 'Auxiliary Boilers Feed Water Header Pressure', 'value': 1300.2970167595486}
{'_id': 'Heat Recovery System Header Mass Flow', 'value': 321470.4298584722}
{'_id': 'Heat Recovery System Header Mass Flow.1', 'value': 325223.13832798606}
{'_id': 'Heat Recovery System Header Pressure', 'value': 1747.4168488828125}
{'_id': 'Heat Recovery System Header Pressure.1', 'value': 1750.4123681006943}
{'_id': 'Lube Oil Cooler Outlet Temperature', 'value': 88.3517489645104}
{'_id': 'Lube Oil Tank Temperature', 'value': 89.94387509161805}
{'_id': 'Motor Current Phase A', 'value': 150.64978842192707}
{'_id': 'Motor C

{'_id': 'Pump Suction Temperature', 'value': 113.17840664157987}
{'_id': 'Pump Thrust Bearing Temperature 1', 'value': 79.08032970557292}
{'_id': 'Pump Thrust Bearing Temperature 2', 'value': 78.94769422345138}
{'_id': 'Auxiliary Boilers A/B Feed Water Header Pressure 2', 'value': 1300.1023006692708}
{'_id': 'Auxiliary Boilers Feed Water Header Pressure', 'value': 1300.1056597121528}
{'_id': 'Heat Recovery System Header Mass Flow', 'value': 332653.9781235764}
{'_id': 'Heat Recovery System Header Mass Flow.1', 'value': 330135.1159535243}
{'_id': 'Heat Recovery System Header Pressure', 'value': 1738.0385376725694}
{'_id': 'Heat Recovery System Header Pressure.1', 'value': 1741.9031393248265}
{'_id': 'Lube Oil Cooler Outlet Temperature', 'value': 70.40618745090453}
{'_id': 'Lube Oil Tank Temperature', 'value': 68.16863123444271}
{'_id': 'Motor Current Phase A', 'value': 0.0}
{'_id': 'Motor Current Phase B', 'value': 0.0}
{'_id': 'Motor Current Phase C', 'value': 0.0}
{'_id': 'Motor Input 

{'_id': 'Pump Thrust Bearing Temperature 1', 'value': 91.34544428987154}
{'_id': 'Pump Thrust Bearing Temperature 2', 'value': 78.55890559757813}
{'_id': 'Auxiliary Boilers A/B Feed Water Header Pressure 2', 'value': 1299.9773756956597}
{'_id': 'Auxiliary Boilers Feed Water Header Pressure', 'value': 1300.0074704685765}
{'_id': 'Heat Recovery System Header Mass Flow', 'value': 317626.2604520833}
{'_id': 'Heat Recovery System Header Mass Flow.1', 'value': 319885.04853317706}
{'_id': 'Heat Recovery System Header Pressure', 'value': 1749.380325980382}
{'_id': 'Heat Recovery System Header Pressure.1', 'value': 1752.861199178125}
{'_id': 'Lube Oil Cooler Outlet Temperature', 'value': 57.57250821455208}
{'_id': 'Lube Oil Tank Temperature', 'value': 59.042978027769095}
{'_id': 'Motor Current Phase A', 'value': 151.27778712930555}
{'_id': 'Motor Current Phase B', 'value': 151.2927960527257}
{'_id': 'Motor Current Phase C', 'value': 149.71182651835068}
{'_id': 'Motor Input Power', 'value': nan}

{'_id': 'Auxiliary Boilers Feed Water Header Pressure', 'value': 1300.468609576736}
{'_id': 'Heat Recovery System Header Mass Flow', 'value': 325879.2293431076}
{'_id': 'Heat Recovery System Header Mass Flow.1', 'value': 328620.1598882986}
{'_id': 'Heat Recovery System Header Pressure', 'value': 1708.3032475371529}
{'_id': 'Heat Recovery System Header Pressure.1', 'value': 1712.082253671354}
{'_id': 'Lube Oil Cooler Outlet Temperature', 'value': 58.863948383545136}
{'_id': 'Lube Oil Tank Temperature', 'value': 61.80367155139584}
{'_id': 'Motor Current Phase A', 'value': 156.10695888199655}
{'_id': 'Motor Current Phase B', 'value': 155.97510802458334}
{'_id': 'Motor Current Phase C', 'value': 154.90166700694445}
{'_id': 'Motor Input Power', 'value': nan}
{'_id': 'Motor Power Factor', 'value': nan}
{'_id': 'Motor Voltage', 'value': 4250.446103765798}
{'_id': 'Pump Discharge Pressure', 'value': 1723.154751192708}
{'_id': 'Pump Discharge Volumetric Flow', 'value': 1001.2808284986979}
{'_id

{'_id': 'Motor Current Phase C', 'value': 148.25895510486112}
{'_id': 'Motor Input Power', 'value': nan}
{'_id': 'Motor Power Factor', 'value': nan}
{'_id': 'Motor Voltage', 'value': 4235.673635117709}
{'_id': 'Pump Discharge Pressure', 'value': 1771.3548972055555}
{'_id': 'Pump Discharge Volumetric Flow', 'value': 905.5019021878126}
{'_id': 'Pump Journal 1 Bearing Temperature', 'value': 145.82463574166667}
{'_id': 'Pump Journal 2 Bearing Temperature', 'value': 148.68148931118054}
{'_id': 'Pump Radial Bearing Vibration', 'value': 7.620395766358159}
{'_id': 'Pump Shaft Speed', 'value': 3588.7450062164926}
{'_id': 'Pump Suction Pressure 1', 'value': 24.723137031342013}
{'_id': 'Pump Suction Pressure 2', 'value': 16.987690503239584}
{'_id': 'Pump Suction Strainer Differential Pressure', 'value': 0.3157439783171875}
{'_id': 'Pump Suction Temperature', 'value': 226.7749322974653}
{'_id': 'Pump Thrust Bearing Temperature 1', 'value': 110.07992695637154}
{'_id': 'Pump Thrust Bearing Temperatu

{'_id': 'Motor Current Phase C', 'value': 147.7952481801389}
{'_id': 'Motor Input Power', 'value': 994.6650619194445}
{'_id': 'Motor Power Factor', 'value': 0.909999967}
{'_id': 'Motor Voltage', 'value': 4238.4964848125}
{'_id': 'Pump Discharge Pressure', 'value': 1775.1878545059028}
{'_id': 'Pump Discharge Volumetric Flow', 'value': 896.2169970834722}
{'_id': 'Pump Journal 1 Bearing Temperature', 'value': 149.68493607671874}
{'_id': 'Pump Journal 2 Bearing Temperature', 'value': 153.61382268159724}
{'_id': 'Pump Radial Bearing Vibration', 'value': 7.700995004310417}
{'_id': 'Pump Shaft Speed', 'value': 3589.198583877604}
{'_id': 'Pump Suction Pressure 1', 'value': 24.66574424443924}
{'_id': 'Pump Suction Pressure 2', 'value': 16.944259001779514}
{'_id': 'Pump Suction Strainer Differential Pressure', 'value': 0.318574618815625}
{'_id': 'Pump Suction Temperature', 'value': 227.0769379224479}
{'_id': 'Pump Thrust Bearing Temperature 1', 'value': 119.08327766487847}
{'_id': 'Pump Thrust B

{'_id': 'Motor Power Factor', 'value': 0.9105767325942707}
{'_id': 'Motor Voltage', 'value': 4236.3832286625}
{'_id': 'Pump Discharge Pressure', 'value': 1764.0511434859377}
{'_id': 'Pump Discharge Volumetric Flow', 'value': 923.5130114350001}
{'_id': 'Pump Journal 1 Bearing Temperature', 'value': 138.68960327593751}
{'_id': 'Pump Journal 2 Bearing Temperature', 'value': 144.61788699256945}
{'_id': 'Pump Radial Bearing Vibration', 'value': 8.20576882481198}
{'_id': 'Pump Shaft Speed', 'value': 3590.5902756489586}
{'_id': 'Pump Suction Pressure 1', 'value': 24.369494268338542}
{'_id': 'Pump Suction Pressure 2', 'value': 16.859038238407987}
{'_id': 'Pump Suction Strainer Differential Pressure', 'value': 0.3334798676855903}
{'_id': 'Pump Suction Temperature', 'value': 226.9739028985417}
{'_id': 'Pump Thrust Bearing Temperature 1', 'value': 102.96813309416666}
{'_id': 'Pump Thrust Bearing Temperature 2', 'value': 93.4169794956493}
{'_id': 'Auxiliary Boilers A/B Feed Water Header Pressure 2

In [41]:
agg_result